src/error.rs

```diff
@@ -13,6 +14,8 @@ pub enum Error {
     SerdeJson(#[cause] serde_json::Error),
     #[fail(display = "{}", _0)]
     SerdeToml(#[cause] toml::de::Error),
+    #[fail(display = "Acquiring lock failed")]
+    PoisonedLockError,
```

Comment:
Okay so for this make it `PoisonedLockError(#[cause] PoisonError)`
It's a wrapper for the inner error and the `#[cause]` tells failure how to get it

Reply:
I tried it that way first but this is where I got stuck because `PoisonError` is generic. So if I write exactly that the type parameter is missing. Not sure how to fix that.